### ISSUE TITLE
Problem with a property of an action homomorphism

### ISSUE THREAD

**Issue by ThomasBreuer** [gap-system/gap#1971]
The following happens in GAP 4.8.7, 4.8.8  and a not too old version of the master branch:

```
gap> g:= SymmetricGroup( 4 );;
gap> tr:= RightTransversal( g, Group( (1,2) ) );;
gap> hom:= ActionHomomorphism( g, tr, OnRight );;
gap> IsSurjective( hom );;
gap> IsInjective( hom );
Error, Value property is already set the other way in
  SetIsTotal( mapi, IsSurjective( hom ) 
 ); at /gap/gap4r8p8/gap4r8/lib/ghom.gi:671 called from 
InverseGeneralMapping( hom 
 ) at /gap/gap4r8p8/gap4r8/lib/ghom.gi:807 called from
attr( AsGroupGeneralMappingByImages( hom ) 
 ) at /gap/gap4r8p8/gap4r8/lib/ghom.gd:274 called from
<function "unknown">( <arguments> )
 called from read-eval loop at line 5 of *stdin*
you can 'return;' to set it anyhow
```

This has worked in older versions of GAP.

**Comment by hulpke** [comment]
This seems to stem from using `Image` instead of `Range` in line 668 of ghom.gi (indicating once more that we want not just inverse but an `inverseRestrictedToImage'...) I'll see whether I can just change this one line without problems cropping up elsewhere.

**Comment by fingolfin** [comment]
Rhis reminds me of issue #944, which I kind of forgot...

**Comment by hulpke** [comment]
Ok, I've looiked at this again, as there was some years ago an issue where code implicitly wanted an inverse defined on the image. So just flipping it back is likely to trigger problems elsewhere.
Currently restricting a homomorphism is comparatively expensive. (Yes, I still have homomorphism cleanup on my to-do list :-( )

What I would really like to do is to add a separate operation `RestrictedInverse` or `TotalInverse` that forms the inverse restricted to the image and then go through the library and checke where `Inverse' is called, and whether it implicitly needs this other version of inverse.

a) Is it wise/desireable to do such a change now? If so:

b) I reckon I could get this done by end of next week, probably earlier. Is this too much of a delay?

c) (This is *not* for 4.9): For historical reasons ActionHomomorphisms have the range S_m, unless they are forced to be surjective. Once the degree get bigger than a few 10's there is no purpose in this, and the large S_m is just a millstone around the neck. Would it be plausible to change this default behavior for 4.10, i.e. make action homomorphisms surjective, unless it is explicitly demanded to be into S_m. What would be the correct way of doing so?

**Comment by fingolfin** [comment]
This has been fixed by 3fc2ed3072c67ba917d441807f3428a21b5ed2f1.

I'll migrate @hulpke's question to a new issue.
